### PR TITLE
Add diagram of installer-based provisioning with full-host bootdisk

### PR DESCRIPTION
**Pull request by Lennonka**
#### What changes are you introducing?

Adding a sequence diagram of installer-based provisioning with full-host bootdisk. Using PlantUML to build the diagram.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

Preparation for improvements of _Provisioning hosts_. The diagram will be used later to explain the workflow of the provisioning method.

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

* Some files (.iuml, .pstyle) are expected to be reused in mutiple diagrams.
* Puppet steps are not currenly displayed in the resulting image. Perhaps they should be for Foreman?
* Continuation of https://github.com/theforeman/foreman-documentation/pull/3069

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into: N/A

### CODE REVIEW

**Comment by Lennonka** [comment]
@stejskalleos GH won't let me to assign you as a reviewer to this PR, so I'm tagging you :)

**Comment by Lennonka** [comment]
Initial feedback incorporated. cc @stejskalleos

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+Proxy -> Proxy : Deploy bootloader configuration files\nto ""/var/lib/tftpboot""
+Foreman -> Proxy : Command to download installer kernel\nand initial RAM disk
+Proxy -> Proxy : Download kernel and initial RAM disk
+Foreman -> Proxy : Create DNS records
+Proxy -> DNS : Forward DNS records
+Proxy -> Foreman : Report status of TFTP, DHCP & DNS
+Foreman -> User : Host is created\n(build mode enabled)
+
+User -> Foreman : Download the boot-disk ISO of the host
+User -> User : Write the boot-disk ISO\nto an external storage device
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:35]: Do we need to mention it? I would be fine just to have `Host -> Host : Boot from the Full host iso` on the line 42.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+User -> Host : Configure to boot\nfrom the external storage device
+
+== Boot into OS installer ==
+
+User -> Host : Power on
+Host -> Host : Use the IP address from the boot disk
+Host -> Host : Boot from the external storage device
+Host -> Host : Load operating system installer
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:43]: We should also mention here that kernel and initramdisk are also loaded from the ISO.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+User -> Host : Eliminate the external storage device\n(too soon?)
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:44]: TBH, I don't know when it's happening, but is it necessary to mention it?

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+!if ($puppet)
+participant "Puppet\nserver" as Puppet
+!endif
+
+note over Host : Powered off
+
+== Create host in Foreman ==
+
+User -> Foreman : Create host
+User -> Foreman : Assign a static IP address
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:22]: This is true only when the Foreman proxy is not managing the DHCP
```
foreman-installer --foreman-proxy-dhcp false --foreman-proxy-dhcp-managed false
```

With managed DHCP Foreman still ask Proxy for free IPs.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:1]: Let's split this workflow into two diagrams.

1. With managed DHCP/DNS
2. Unmanaged DHCP/DNS

The main benefit of the Full Host ISO is its content. It already has the kernel, init RAM disk, Grub2 config file, and Foreman's URL - all in one ISO. Example:
```
├── BOOT
│   ├── _MIRROR_GXRRL2UPF2NL_VMLINUZ <- Downloaded from the inst. media
│   └── RROR_GXRRL2UPF2NL_INITRD_IMG <- Downloaded from the inst. media
├── boot.cat
├── EFI
│   └── BOOT
│       └── grub.cfg <- Generated by Foreman
├── efiboot.img
├── grub-hdd.cfg
├── isolinux.bin
├── isolinux.cfg <- Generated by Foreman
├── ldlinux.c32
├── libutil.c32
└── menu.c32
```

Technically, the main benefit is that you need a proxy to serve these files but only generate the kickstart template.

During the host creation process, the user adds a MAC address and (random) IP address, and the provisioned host takes the real IP from the DHCP server during the installation, later reporting back to Foreman the correct IP from the facts.

**Review by stejskalleos — CHANGES_REQUESTED** [review]

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 2 above)
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:43]: That's the kernel and init RD of the installer, right? I'll rephrase it to be more specific.

**Comment by Lennonka** [comment]
Summary of today's call:
- We agreed to create this diagram for unmanaged DHCP only.
- Leos will check when and how the elimintation of the external storage device happens.

cc @stejskalleos

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 1 above)
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:41]: Yes, we do because the step is relevant from user's perspective.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+Foreman -> Proxy : Command to download installer kernel\nand initial RAM disk
+Proxy -> Proxy : Download kernel and initial RAM disk
+Foreman -> Proxy : Create DNS records
+Proxy -> DNS : Forward DNS records
+Proxy -> Foreman : Report status of TFTP, DHCP & DNS
+Foreman -> User : Host is created\n(build mode enabled)
+
+User -> Foreman : Download the full boot-disk ISO of the host
+Foreman -> Foreman : Generate the full boot-disk ISO of the host
+Foreman -> User : Return the full boot-disk ISO of the host
```
Comment (Lennonka) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:40]: @stejskalleos Like this or do we want to go into more detail?

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 3 above)
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:50]: We agreed that Leos will investigate when and how this happens.

**Comment by Lennonka** [comment]
@stejskalleos You mentioned that the host reports its actual IP address with other facts if I remember correctly. I'm not sure if it applies to unmanaged DHCP. If it does, when exactly does it happen?

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+!if ($puppet)
+participant "Puppet\nserver" as Puppet
+!endif
+
+note over Host : Powered off
+
+== Create host in Foreman ==
+
+User -> Foreman : Create host
+User -> Foreman : Assign a temporary IP address
```
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:26]: "assign temporary IP" is not needed. Replace with assign a MAC address

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 7]
```diff
@@ -0,0 +1,69 @@
+@startuml
+
+!include foreman.pstyle
+!$networkboot = 0
+
+title Installer-based provisioning with full-host boot disk on bare metal
```
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:6]: Actually, we have a [request](https://github.com/theforeman/foreman-documentation/issues/619) to document Bootdisk+VMware :) So yes, that's a good point.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 3 above)
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:50]: Updated.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+note over Host : Powered off
+
+== Create host in Foreman ==
+
+User -> Foreman : Create host
+User -> Foreman : Assign an MAC address
+Foreman -> Proxy : Create DNS records
+Proxy -> DNS : Forward DNS records
+Proxy -> Foreman : Report status of TFTP, DHCP & DNS
```
Comment (Lennonka) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:29]: @stejskalleos Perhaps we can reduce this when we aren't using DHCP and TFTP?
```suggestion
Proxy -> Foreman : Report status of DNS
```

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+Foreman -> User : Host is created\n(build mode enabled)
+
+User -> Foreman : Request to generate the full host boot-disk ISO
+group Template [Boot disk iPXE - host]
```
Comment (Lennonka) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:33]: @stejskalleos Is this the correct template in this scenario?

Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:33]: Check Settings

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+User -> Host : Power on
+Host -> Host : Use the IP address from the boot disk
+Host -> Host : Boot from the external storage device
+Host -> Host : Load installer kernel\nand initial RAM disk from the boot disk
+User -> Host : Eliminate the external storage device\n(too soon?)
+!include prov-installation-media.iuml
+note over Host : Operating system installed
+!include prov-initial-configuration.iuml
+Host -> Foreman : Call home\n(disables build mode)
```
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:52]: Host -> Foreman : Update IP address from Facts
OR
Call home
Foreman -> Foreman : Update IP address from the 'built' request

**Comment by maximiliankolb** [comment]
from triage: @stejskalleos Please also have a look at this PR. Thank you.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+Proxy -> Foreman : Report status of DNS
+Foreman -> User : Host is created\n(build mode enabled)
+
+User -> Foreman : Request to generate the full host boot-disk ISO
+group Templates [Boot disk iPXE - host, PXELinux, PXEGrub2]
+    Foreman -> Proxy : Render boot-loader configuration files
+end
+Foreman -> Proxy : Download kernel and initial RAM disk
+Foreman -> Foreman : Add the boot loader to ISO files
+Foreman -> Foreman : Build the full host boot-disk ISO
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:38]:
```suggestion
Foreman -> Foreman : Add PXELinux & PXEGrub2 config files
Foreman -> Foreman : Build the full host boot-disk ISO
```

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+Foreman -> User : Download the full host boot-disk ISO
+User -> User : Write the boot-disk ISO\nto an external storage device
+User -> Host : Configure to boot\nfrom the external storage device
+
+== Boot into OS installer ==
+
+User -> Host : Power on
+Host -> Host : Use the IP address from the boot disk
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:46]: Only if the static IP has been added during the host creation,  or we are in an unmanaged environment.
Otherwise, the machine will get an IP from the DHCP.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+Host -> Host : Boot from the external storage device
+Host -> Host : Load installer kernel\nand initial RAM disk from the boot disk
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:48]:
```suggestion
Host -> Host : Load installer kernel\nand initial RAM disk from the boot disk
Host -> Host : Load config files for bootloader from the boot disk
```

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 14]
```diff
@@ -0,0 +1,73 @@
+@startuml
+
+!include foreman.pstyle
+!$networkboot = 0
+
+title Installer-based provisioning with full-host boot disk on bare metal
+
+left header
+In the described scenario, Foreman does not manage DHCP.
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:9]: Where is this header displayed? I don't see it in the picture

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 12 above)
Comment (stejskalleos) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:46]: I just saw the _header_ section, somehow I missed it in the picture.

**Review by stejskalleos — CHANGES_REQUESTED** [review]

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 14 above)
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:9]: Top left corner. Unfortunately, the font is too small. I'll try to find out if I can make it larger.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 13 above)
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:48]: Are you sure about the order of these steps?
To me, it seems more logical that the host loads:
1. The bootloader
2. Bootloader config
3. Installer kernel etc.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 11 above)
Comment (Lennonka) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:38]: Makes sense, however, elsewhere we've been calling these files "boot-loader configuration" -- is that ok?

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+User -> Host : Power on
+Host -> Host : Use the IP address from the boot disk
+Host -> Host : Boot from the external storage device
+Host -> Host : Load installer kernel\nand initial RAM disk from the boot disk
+User -> Host : Eliminate the external storage device\n(too soon?)
+!include prov-installation-media.iuml
+note over Host : Operating system installed
+!include prov-initial-configuration.iuml
+Host -> Foreman : Call home\n(disables build mode)
+Host -> Host : Reboot
```
[the remark this answers is not in the document]
Comment (stejskalleos) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:58]: ACK.

Or we can say shutdown the host, remove the device, boot. Either way is fine.

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 11 above)
Comment (stejskalleos) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:38]: ACK

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml (hunk 13 above)
Comment (stejskalleos) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:48]: Yes, you are right, your order is correct

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 16]
```diff
@@ -35,6 +34,8 @@ group Templates [Boot disk iPXE - host, PXELinux, PXEGrub2]
 end
 Foreman -> Proxy : Download kernel and initial RAM disk
 Foreman -> Foreman : Add the boot loader to ISO files
+Foreman -> Foreman : Add the boot-loader configuration files to ISO files
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:37]:
```suggestion
Foreman -> Foreman : Add the boot-loader configuration files to ISO file
```

File: guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml [hunk 17]
```diff
@@ -45,6 +46,8 @@ User -> Host : Configure to boot\nfrom the external storage device
 User -> Host : Power on
 Host -> Host : Use the IP address from the boot disk
```
Comment (stejskalleos) [inline guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:47]:
```suggestion
```

We don't store the IP address in the ISO. The IP address is stored in the DHCP server, and it's provided by the DHCP server during the provisioning time.

Comment (stejskalleos) [reply guides/image-sources/provisioning-installer-bootdisk-fullhost.plantuml:47]: There should be a same step as in https://github.com/theforeman/foreman-documentation/pull/3309, Getting the IP address from the dhcp server

File: guides/image-sources/prov-first-local-boot-hdd.iuml
Comment (Lennonka) [inline guides/image-sources/prov-first-local-boot-hdd.iuml:1]: @stejskalleos Please, check this. These steps should represent what happens at the First local boot.

**Comment by Lennonka** [comment]
Style review has been incorporated from another PR.
